### DISCUSSION TITLE
LF/CRLF windows/linux

### DISCUSSION THREAD

**Discussion by mwcarlis** [morrownr/8814au#113] [General]
I can't be the only person who runs into this can I?  Was trying to build/install the module and all the bash scripts in the repo were failing for no apparent reason....  A couple of minute later I realized there is CRLF encoding in them?  Is someone editing files in windows?

First thing I saw was the following when running install-driver.sh.
`/bin/bash^M: bad interpreter: No such file or directory
`

Looks like this is what we get in the repo?  note: 0x0d, 0x0a (this is not actually install-driver.sh, but it serves the example)
```
@:~/8814au$ hexdump -C dkms-make.sh 
00000000  23 21 2f 62 69 6e 2f 73  68 0d 0a 0d 0a 23 20 53  |#!/bin/sh....# S|
00000010  4d 45 4d 20 6e 65 65 64  73 20 74 6f 20 62 65 20  |MEM needs to be |
```
A few seds later.  now: 0x0a 0x0a  
```
@:~/8814au$ hexdump -C dkms-make.sh 
00000000  23 21 2f 62 69 6e 2f 73  68 0a 0a 23 20 53 4d 45  |#!/bin/sh..# SME|
00000010  4d 20 6e 65 65 64 73 20  74 6f 20 62 65 20 73 65  |M needs to be se|
```
After I made install-driver.sh use LF encoding I then hit something like this, but it turned out I also had to LF dkms.conf. After that things seem to work. 
```Running install-driver.sh version 20210917
Starting installation...
Copying source files to: /usr/src/rtl8814au-5.8.5.1
Copying 8814au.conf to: /etc/modprobe.d

Creating symlink /var/lib/dkms/rtl8814au/5.8.5.1/source ->
                 /usr/src/rtl8814au-5.8.5.1

DKMS: add completed.

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
...(bad exit status: 2)78-generic KSRC=/lib/modules/5.15.0-78-generic/build
ERROR (dkms apport): binary package for rtl8814au: 5.8.5.1 not found
Error! Bad return status for module build on kernel: 5.15.0-78-generic (x86_64)
Consult /var/lib/dkms/rtl8814au/5.8.5.1/build/make.log for more information.
An error occurred. dkms build error = 10
Please report this error.
troll@:~/8814au$ 
```

I have to say, this is a little bit of a trap because scripts call other scripts & its hard to tell exactly which file is complaining..  My best guess is that someone that files are edited in a linux VM, but in host windows filesystem?
`5.15.0-78-generic #85~20.04.1-Ubuntu SMP Mon Jul 17 09:42:39 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
`

**Comment by morrownr** [comment]
Hi @mwcarlis 

> I can't be the only person who runs into this can I?

Possibly so.

> A couple of minute later I realized there is CRLF encoding in them?

I wonder how that happened.

> Is someone editing files in windows?

Highly unlikely as this is my repo and I have never been a Windows user. I use geany for editing the scripts.

I just checked the repo. All I am finding is LF in the scripts.

This repo averages around 100 clones per day. I have not seen this reported before. You might think about doing a fresh test with a new clone.

In case you are not aware of it, the Main Menu for this site is located at:

 https://github.com/morrownr/USB-WiFi

There is a lot of info that might be of interest.

Regards

**Reply by mwcarlis** [reply]
Hi @morrownr 

> I just checked the repo. All I am finding is LF in the scripts.

Very interesting.. I am throughly confused then. Will see about it later tonight when back at my workstation. I had even reset hard to main.

**Reply by mwcarlis** [reply]
@morrownr It looks like my problem is that somehow this became set in the .gitconfig on this system. Thanks for getting back on this so quickly... 
`autocrlf = true`
